src/Sarif/IAnalysisLogger.cs

```diff
@@ -24,7 +24,10 @@ public interface IAnalysisLogger
         /// <param name="notification">
         /// The notification to log.
         /// </param>
-        void LogToolNotification(Notification notification);
+        /// <param name="associatedRule">
+        /// The scan rule, if any, associated with the notification.
+        /// </param>
+        void LogToolNotification(Notification notification, ReportingDescriptor associatedRule = null);
```

Comment:
>associatedRule = null [](http://example.com/codeflow?start=80&length=21)

An optional argument in an interface contract is an interesting thing. Any caller to the interface does not need to provide this argument. Anyone implementing this interface can override the optional value.